x-pack/test_serverless/functional/test_suites/common/platform_security/roles.ts

```diff
@@ -36,41 +41,54 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
   const log = getService('log');
 
   describe('Roles', function () {
-    before(async () => {
-      await pageObjects.svlCommonPage.loginAsAdmin();
-      await pageObjects.common.navigateToApp('management');
-      await pageObjects.svlManagementPage.assertRoleManagementCardExists();
-      await pageObjects.svlManagementPage.clickRoleManagementCard();
-      const url = await browser.getCurrentUrl();
-      expect(url).to.contain('/management/security/roles');
-    });
+    describe('as Viewer', () => {
+      before(async () => {
+        await pageObjects.svlCommonPage.loginAsViewer();
+        await pageObjects.common.navigateToApp('management');
+      });
 
-    after(async () => {
-      await clearAllRoles(es, log);
+      it('should not display the roles management card', async () => {
+        await pageObjects.svlManagementPage.assertRoleManagementCardDoesNotExist();
+      });
     });
 
-    it('should not display reserved roles', async () => {
-      const table = await testSubjects.find('rolesTable');
-      const tableContent = await table.getVisibleText();
-      expect(tableContent).to.contain('No custom roles to show');
-    });
+    describe('as Admin', () => {
+      before(async () => {
+        await pageObjects.svlCommonPage.loginAsAdmin();
+        await pageObjects.common.navigateToApp('management');
+        await pageObjects.svlManagementPage.assertRoleManagementCardExists();
+        await pageObjects.svlManagementPage.clickRoleManagementCard();
+        const url = await browser.getCurrentUrl();
+        expect(url).to.contain('/management/security/roles');
+      });
+
+      after(async () => {
+        await clearAllRoles(es, log);
+      });
 
-    it('should create and only display custom roles', async () => {
-      const customRole = 'serverless-custom-role';
-      await pageObjects.security.addRole(customRole, {
-        elasticsearch: {
-          indices: [
-            {
-              names: ['dlstest'],
-              privileges: ['read', 'view_index_metadata'],
-            },
-          ],
-        },
+      it('should not display reserved roles', async () => {
+        const table = await testSubjects.find('rolesTable');
+        const tableContent = await table.getVisibleText();
+        expect(tableContent).to.contain('No custom roles to show');
       });
 
-      const table = await testSubjects.find('rolesTable');
-      const tableContent = await table.getVisibleText();
-      expect(tableContent).to.contain(customRole);
+      it('should create and only display custom roles', async () => {
```

Comment:
**nit:** I'm nitpicking here, but technically this doesn't verify that we **only** display custom roles. It just checks that we display the role we created, and if I understand correctly, it wouldn't fail if other roles were displayed (e.g., if there's a bug and we show reserved roles as long as there's at least one non-reserved role) :)

Reply:
Good point. I will update.